### ISSUE TITLE
`typedef uint8_t<_Iftrue, _Iffalse> AutocxxConcrete0;`

### ISSUE THREAD

**Issue by adetaylor** [google/autocxx#470]
Reproduce per #468.

Get

```
[autocxx-s2-example 0.1.0] cargo:warning=/usr/local/google/home/adetaylor/autocxx/examples/s2/target/debug/build/autocxx-s2-example-e7b8302a7b539eeb/out/autocxx-build-dir/include/autocxxgen.h:10:9: error: ‘uint8_t’ {aka ‘unsigned char’} is not a template
[autocxx-s2-example 0.1.0] cargo:warning=   10 | typedef uint8_t<_Iftrue, _Iffalse> AutocxxConcrete0;
[autocxx-s2-example 0.1.0] cargo:warning=      |         ^~~~~~~
```

**Comment by adetaylor** [comment]
bindgen gives us:

```rust
[autocxx-s2-example 0.1.0]             #[bindgen_original_name("Invoker::type")]
[autocxx-s2-example 0.1.0]             pub type Invoker_type = u8<_Iftrue, _Iffalse>;
```

Appears to be in `absl::base_internal`.

**Comment by adetaylor** [comment]
Presumably from

```cpp

// Resolves to the first matching clause.
   template <typename... Args>
   struct Invoker {
   typedef typename std::conditional<
   MemFunAndRef::Accept<Args...>::value, MemFunAndRef,
   typename std::conditional<
   MemFunAndPtr::Accept<Args...>::value, MemFunAndPtr,
   typename std::conditional<
   DataMemAndRef::Accept<Args...>::value, DataMemAndRef,
   typename std::conditional<DataMemAndPtr::Accept<Args...>::value,
   DataMemAndPtr, Callable>::type>::type>::
   type>::type type;
  };
```

**Comment by adetaylor** [comment]
creduced code:

```cpp
namespace std {
template <bool, typename _Iftrue, typename _Iffalse> struct a;
}
namespace {
namespace {
template <typename> struct b { template <typename> struct c; };
struct d : b<d> {};
struct e : b<e> {};
struct f : b<f> {};
struct g : b<g> {};
struct h;
template <typename... i> struct j {
  typedef std::a<d::c<i...>::k, d,
                 std::a<e::c<i...>::k, e,
                        std::a<f::c<i...>::k, f, std::a<g::c<i...>::k, g, h>>>>
      l;
};
} // namespace
} // namespace
```

**Comment by adetaylor** [comment]
Raised https://github.com/rust-lang/rust-bindgen/issues/2048 though I'm not yet sure if bindgen is WAI.

**Comment by adetaylor** [comment]
I have a plan here. I think our bindgen blocklist is now unnecessarily broad, now we're good at ignoring types which we can't handle. I am working on narrowing it.

**Comment by adetaylor** [comment]
This bug was fixed by #481, but the same test case now gives us #492.
